test/TesterInternal/StorageTests/PersistenceGrainTests.cs

```diff
@@ -1200,45 +1177,58 @@ private void SetErrorInjection(string providerName, ErrorInjectionPoint errorInj
                 //}
             }
         }
-
-        private MockStorageProvider FindStorageProviderInUse(string providerName, bool okNull = false)
+        private bool CanGetStorageProviderInUse(string providerName)
         {
-            MockStorageProvider providerInUse = null;
-            SiloHandle siloInUse = null;
-            List<SiloHandle> silos = this.HostedCluster.GetActiveSilos().ToList();
-            foreach (var siloHandle in silos)
+            foreach (var siloHandle in this.HostedCluster.GetActiveSilos())
             {
-                MockStorageProvider provider = (MockStorageProvider)siloHandle.AppDomainTestHook.GetStorageProvider(providerName);
-                Assert.NotNull(provider);
-                if (provider.ReadCount > 0)
+                if (siloHandle.TestHook.CanGetStorageProvider(providerName).Result)
                 {
-                    Assert.Null(providerInUse);
-                   
-                    providerInUse = provider;
-                    siloInUse = siloHandle;
```

Comment:
this is different from what it used to do before, are tests still passing? basically this was called FindStorageProvider *InUse* because it would check which one was being used from all the silos in the cluster. The current assertion is just checking that the provider exists. If the current assertion is enough, consider renaming the method or skipping it altogether and just favor the method below all times (`GetStateForStorageProviderInUse`)